design/ux/Obscuro_Gateway.md

```diff
@@ -0,0 +1,264 @@
+    database "Alice's Viewing Keys"
+}
+Alice --> "Alice's MetaMask"
+"Alice's MetaMask" --> "Alice's Wallet Extension"
+"Alice's Wallet Extension" <-> "Alice's Viewing Keys"
+"Alice's Wallet Extension" ----> "Obscuro Nodes" : Encrypted RPC
+
+actor Bob
+component "Bob's Computer"{
+    agent "Bob's MetaMask"
+}
+component "Confidential Web Service"{
+    node "Obscuro Gateway"
+    database "OG Viewing Keys"
+}
+"OG Viewing Keys" <-> "Obscuro Gateway"
+
+Bob --> "Bob's MetaMask"
+"Bob's MetaMask" ---> "Obscuro Gateway" : HTTPS
+"Obscuro Gateway" ----> "Obscuro Nodes" : Encrypted RPC
+
+actor Charlie
+component "Charlie's Computer"{
+    agent "Charlie's MetaMask"
+}
+node "Obscuro Gateway"
+Charlie --> "Charlie's MetaMask"
+"Charlie's MetaMask" ---> "Obscuro Gateway" : HTTPS
+
+@enduml
+```
+
+Notice that the OG is a multi-tenant WE running inside SGX and storing the authenticated viewing keys (and other information) in an encrypted database.
+
+
+## High level UX
+
+The key reason for the OG is to allow implementing a 3-click user onboarding process.
+
+See below for a proposed flow:
+![UX diagram](resources/og_ux.png)
+
+```plantuml
+@startuml
+'https://plantuml.com/sequence-diagram
+
+autonumber
+
+actor "Alice's Browser" as Alice
+participant MetaMask as MM
+participant "https://gateway.obscuro.network/v1" as OG
+participant "https://obscuro.network" as ON
+
+group First click
+    Alice -> ON: Join Obscuro
+    ON --> Alice: Redirect to OG
+    note right
+        The point of this sequence
+        is to call the "Create User"
+        endpoint on the OG
+    end note
+    Alice -> OG: Create User\n(ajax call behind the scenes)
+    OG -> OG: Generate and record VK\nand record against UserId
+    note right
+        The UserId is the hash of
+         the Public Key of the VK
+    end note
+    OG -> Alice: Send UserId
+    Alice -> MM: Automatically add "Obscuro" network with RPC\n"https://gateway.obscuro.network/v1?u=$UserId"
+end
+
+group Second click
+    Alice -> MM : Connect wallet and switch to Obscuro
+end
+
+group Third click
+    Alice -> MM: Automatically open MM with request to \nsign over "Register $UserId for $Acct"
+    note right
+        This text will be sent as is
+        accompanied by the signature and
+        will be parsed inside the enclave
+    end note
+    Alice -> MM : Confirm signature
+end
+
+Alice -> OG: All further Obscuro interactions will be to\nhttps://gateway.obscuro.network/v1?u=$UserId
+
+@enduml
+```
+
+The onboarding should be done in 3 clicks.
+1. The user goes to a website (like "obscuro.network"), where she clicks "Join Obscuro". This will add a network to their wallet.
+2. User connects the wallet to the page.
+3. In the wallet popup, the user has to sign over a message: "Register $UserId for $ACCT"
+
+##### Click 1
+1. Behind the scenes, a js functions calls "gateway.obscuro.network/v1/join" where it will generate a VK and send back the hash of the Public key. This is the "UserId" 
```

Comment:
```suggestion
1. A js functions calls "gateway.obscuro.network/v1/join" where it will generate a VK and send back the hash of the Public key. This is the "UserId" 
```